handle reordered NEW_CONNECTION_ID frames (#3202)
quicwg/base-drafts: draft-ietf-quic-transport.md

retire the corresponding connection IDs and send the corresponding
RETIRE_CONNECTION_ID frames in a timely manner.

An endpoint that receives a NEW_CONNECTION_ID frame with a sequence number

> smaller than the Retire Prior To field of a NEW_CONNECTION_ID frame received

Reviewer: ```suggestion
smaller than the Retire Prior To field of a previously received NEW_CONNECTION_ID
```

> before, MUST immediately send a corresponding RETIRE_CONNECTION_ID frame.

Reviewer: ```suggestion
frame MUST immediately send a corresponding RETIRE_CONNECTION_ID frame.
```

Reviewer 2: I would potentially also add
```suggestion
MUST immediately send a corresponding RETIRE_CONNECTION_ID frame that 
retires the newly received connection ID.
```
Since there's two NCID frames being referenced here.

Author: Thanks! I applied your suggestions.

> before, it MUST immediately send a corresponding RETIRE_CONNECTION_ID frame.

Reviewer 3: ```suggestion
before, MUST immediately send a corresponding RETIRE_CONNECTION_ID frame.
```

retire the corresponding connection IDs and send the corresponding
RETIRE_CONNECTION_ID frames in a timely manner.

An endpoint that receives a NEW_CONNECTION_ID frame with a sequence number
smaller than the Retire Prior To field of a previously received

> NEW_CONNECTION_ID frame, MUST immediately send a corresponding

Reviewer 3: ```suggestion
NEW_CONNECTION_ID frame MUST immediately send a corresponding
```
